runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java

```diff
@@ -392,4 +408,59 @@ public State waitUntilFinish(Duration duration) throws IOException {
           "DirectPipelineResult does not support waitUntilFinish.");
     }
   }
+
+  /**
+   * Internal configuration options for the {@link DirectRunner}. Deals with internal implementation
+   * details.
+   */
+  public interface InternalOptions extends DirectOptions {
+    /**
+     * Gets the {@link ExecutorServiceFactory} to use to create instances of {@link ExecutorService}
+     * to execute {@link PTransform PTransforms}.
+     *
+     * <p>Defaults to a {@link FixedThreadPoolSupplier}, which produces instances of
+     * {@link Executors#newFixedThreadPool(int)}.
+     */
+    @JsonIgnore
+    @Required
+    @Hidden
+    @Default.InstanceFactory(FixedThreadPoolSupplier.class)
+    OptionsSupplier<ExecutorService> getExecutorServiceFactory();
+
+    void setExecutorServiceFactory(OptionsSupplier<ExecutorService> executorService);
+
+    /**
+     * Gets the {@link Clock} used by this pipeline. The clock is used in place of accessing the
+     * system time when time values are required by the evaluator.
+     */
+    @Default.InstanceFactory(NanosOffsetClock.Factory.class)
```

Comment:
Instead of going through an interface, should we not just have the PTransform override applier modify/interact with the DirectRunner directly since they are part of the same package.

This prevents people from mucking with an internal option of the direct runner.

Ditto for the executor as well.

This might make it more difficult to test though, so need to weigh the trade off